### PR TITLE
dev: Use centralized sync-rtd-redirects workflow

### PR DESCRIPTION
**Pull request by victorlin**
## Description of proposed changes

A central copy is easier to maintain over time now that this workflow is proliferating across our repos.

## Pre-merge

- ~Merge https://github.com/nextstrain/.github/pull/33~ not strictly blocking

## Testing

- [x] PR dry run sync is successful
- [x] Other checks pass

## Release checklist

N/A, internal dev change

### CODE REVIEW

File: .github/workflows/sync-redirects.yaml [hunk 1]
```diff
@@ -13,20 +13,10 @@ on:
 
 jobs:
   sync:
-    # Prevent this job from running on forks.
-    if: github.repository_owner == 'nextstrain'
```
Comment (tsibley) [inline .github/workflows/sync-redirects.yaml:17]: You could preserve this `if` on the job if you wanted, even with the call of the centralized workflow.

**Review by tsibley — APPROVED** [review]
Might be nice to also add:

```yaml
on:
  pull_request:
```

since the centralized workflow now supports dry runs for PRs.

File: .github/workflows/sync-redirects.yaml (hunk 1 above)
Comment (victorlin) [reply .github/workflows/sync-redirects.yaml:17]: Added back in 96dde53db6385d74ee8d8b817d6a779049ce2142

**Comment by victorlin** [comment]
> Might be nice to also add:
> 
> ```yaml
> on:
>   pull_request:
> ```
> 
> since the centralized workflow now supports dry runs for PRs.

Good suggestion! Added c1ced4f72e944be1aaf89b10cbaa5dfddd2a7d7a

**Pull request merged**